### PR TITLE
Add basic support for Amazon MQ

### PR DESCRIPTION
**Pull request by fotto**
##### SUMMARY
This PR contains some basic support for Amazon MQ. It covers

* Managing Amazon MQ configurations
* Managing Amazon MQ (local) users
* Broker reboot

not in scope of the PR
* management of the brokers itself (create, delete)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
Amazon MQ

##### ADDITIONAL INFORMATION

We use those modules to manage our MQ Brokers. The context of usage is

* brokers are created by different means (e.g. terraform)
* we've developed a custom role around those modules
* to manage MQ user credentials we use another local extension of amazon.aws collection (will come as separate PR) that interfaces with AWS SecretsManager
* that role uses the reboot broker feature (part of this PR) to implement a custom handler that reboots a broker whenever a configuration run sees any changes in configuration and/or users

The functionality of the added modules is illustrated in the added test suite (tests/integration/targets/mq) which
basically requires just a running MQ broker to be usuable.

### CODE REVIEW

**Comment by goneri** [comment]
Hi @fotto,

Thank you for this Pull Request. New modules should be created in the https://github.com/ansible-collections/community.aws/ repository. They will potentially be promoted in this repository in the future.
